server/plugin_release.go

```diff
@@ -29,199 +35,406 @@ import (
+
+	githubPluginFilePath, err := downloadAsset(ctx, pluginAsset, tmpFolder)
+	if err != nil {
+		return err
+	}
+
+	// Split plugin into platform specific tars
+	platformPluginFilePaths, err := createPlatformPlugins(repositoryName, tag, githubPluginFilePath, tmpFolder)
+	if err != nil {
+		return err
+	}
 
-	_, _, err := client.Repositories.ListBranches(ctx, Cfg.GithubOrg, repo, nil)
+	// Sign plugin tars and put them in tmpFolder. Signature files are assumed to be <path>.sig
+	err = signPlugins(Cfg, append(platformPluginFilePaths, githubPluginFilePath), tmpFolder)
 	if err != nil {
+		return err
+	}
+
+	// Upload github plugin tar signature to github
+	githubPluginSignatureFilePath := fmt.Sprintf("%s.sig", githubPluginFilePath)
+	if err := uploadFilesToGithub(ctx, client, owner, repositoryName, tag, []string{githubPluginSignatureFilePath}); err != nil {
+		return err
+	}
+
+	// Duplicate github plugin tar signature that follows s3 release bucket naming convention
+	s3PluginSignatureFilepath := filepath.Join(tmpFolder, fmt.Sprintf("%v-%v.tar.gz.sig", repositoryName, tag))
+	if err := os.Symlink(githubPluginSignatureFilePath, s3PluginSignatureFilepath); err != nil {
+		return fmt.Errorf("failed to duplicate signature file err=%w", err)
+	}
+
+	s3Bucket := []string{s3PluginSignatureFilepath}
+	for _, p := range platformPluginFilePaths {
+		s3Bucket = append(s3Bucket, p)
+		s3Bucket = append(s3Bucket, fmt.Sprintf("%s.sig", p))
+	}
+
+	// Upload plugins and signatures to s3 release bucket
+	if err := uploadToS3(ctx, Cfg, s3Bucket); err != nil {
+		return fmt.Errorf("failed to upload to s3 err=%w", err)
+	}
+
+	return nil
+}
+
+// checkRepo checks if repo exists.
+func checkRepo(cfg *MatterbuildConfig, owner, repo string) error {
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: cfg.GithubAccessToken})
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
+
+	if _, _, err := client.Repositories.ListBranches(ctx, owner, repo, nil); err != nil {
 		LogError("No branch found. err=" + err.Error())
 		return fmt.Errorf("Looks like this Repository is not part of the org or does not exist. Repo: %s", repo)
-
 	}
 
 	return nil
 }
 
-func createTag(tag, repo string) error {
-	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: Cfg.GithubAccessToken})
+// getReleaseByTag gets github release by tag.
+func getReleaseByTag(cfg *MatterbuildConfig, owner, repositoryName, tag string) (*github.RepositoryRelease, error) {
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: cfg.GithubAccessToken})
 	tc := oauth2.NewClient(ctx, ts)
-	client = github.NewClient(tc)
+	client := github.NewClient(tc)
 
-	refs, _, err := client.Git.ListRefs(ctx, Cfg.GithubOrg, repo, nil)
+	release, _, err := client.Repositories.GetReleaseByTag(ctx, owner, repositoryName, tag)
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("failed to get release by tag err=%w", err)
```

Comment:
Would it be better with mlog or what is the de facto standard?

Reply:
mlog to just log and not return an error? standard way is to use errors pkg or fmt.Errorf()
https://github.com/golang/go/wiki/CodeReviewComments#error-strings